tools/webpack/plugins/loader-apm-check.mjs

```diff
@@ -0,0 +1,42 @@
+/**
+ * Webpack plugin that checks loader files for any character sequences
+ * that are known to break APM injection. If found, an error is registered
+ * in the compilation and the compilation process will end with a non-zero
+ * status code.
+ */
+export default class NRBALoaderApmCheckPlugin {
+  #pluginName = 'NRBALoaderApmCheckPlugin'
+
+  /**
+   * @param compiler {import('webpack/lib/Compiler.js').default}
+   */
+  apply (compiler) {
+    compiler.hooks.thisCompilation.tap(this.#pluginName, (compilation) => {
+      compilation.hooks.processAssets.tap(
+        {
+          name: this.#pluginName,
+          stage: compilation.compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE + 1,
+          additionalAssets: true
+        },
+        (assets) => {
+          Object.entries(assets)
+            .filter(([assetKey]) => assetKey.indexOf('-loader') > -1 && assetKey.endsWith('.js'))
```

Comment:
Both this & non-ascii: only the `.min.js` that APM uses or does it matter/not matter?